plugins/auth-node/api-report.md

```diff
@@ -22,21 +23,39 @@ export type BackstageUserIdentity = {
   ownershipEntityRefs: string[];
 };
 
+// @public
+export class DefaultIdentityClient implements IdentityApi {
+  // @deprecated
+  authenticate(token: string | undefined): Promise<BackstageIdentityResponse>;
+  static create(options: IdentityClientOptions): DefaultIdentityClient;
+  // (undocumented)
+  getIdentity(req: Request_2): Promise<BackstageIdentityResponse | undefined>;
+}
+
 // @public
 export function getBearerTokenFromAuthorizationHeader(
   authorizationHeader: unknown,
 ): string | undefined;
 
 // @public
+export interface IdentityApi {
+  getIdentity(
+    req: Request_2<any>,
```

Comment:
mmm. Im not sure about this one. If we allow token be provided to the API, it would imply that the token is being extracted and provided by the caller. I think the meaning of my effort here is to allow the IdentityApi implementation decide how to extract the identity from the incoming request regardless of how it does it.

I could maybe see a case for the following interface:

```
type IdentityApiGetIdentityRequest = { request: Request<unknown>, context: Context };
```

Can you point me to the Context interface, there seems to be a few in backstage, Also would you mind expanding on the case for this? I might be missing the point.

Reply:
This is the one I was referring to https://github.com/backstage/backstage/blob/0e9785b260a82c1e4c2530b02d3ab00a186eee72/packages/backend-common/src/context/Contexts.ts#L29

It's not in use yet though we envision it probably starting to get more and more needed as we progress. Just meant as an example here.

We just thought that there might be other situations than "i'm right at the start of specifically an express request handler function" where this functionality is needed.

Reply:
ah right, ok, gotya. We can evolve it over time now that it is an object instead of a request specifically.